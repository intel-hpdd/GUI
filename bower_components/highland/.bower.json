{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}