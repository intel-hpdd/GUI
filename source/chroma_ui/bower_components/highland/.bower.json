{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}