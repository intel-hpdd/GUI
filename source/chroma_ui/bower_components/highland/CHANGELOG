Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
