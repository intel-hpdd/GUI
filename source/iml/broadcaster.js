// @flow

//
// Copyright (c) 2017 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import highland from 'highland';

import type {
  HighlandStreamT,
  errorWrapT
} from 'highland';


type streamFnT = () => HighlandStreamT<any>;

export default function broadcaster (source$:HighlandStreamT<any>):streamFnT {
  var latest:mixed;

  const viewers = [];

  source$
    .errors(error => {
      const err:errorWrapT = {
        __HighlandStreamError__: true,
        error
      };

      viewers.forEach(v => v.write(err));
    })
    .each((xs:mixed) => {
      latest = xs;

      viewers.forEach(v => v.write(xs));
    });

  const fn = () => {
    const viewer$:HighlandStreamT<any> = highland()
      .onDestroy(() => {
        const idx = viewers
          .indexOf(viewer$);

        if (idx !== -1)
          viewers.splice(idx, 1);
      });

    if (latest)
      viewer$.write(latest);

    viewers.push(viewer$);

    return viewer$;
  };

  fn.endBroadcast = () => source$.destroy();

  return fn;
}
