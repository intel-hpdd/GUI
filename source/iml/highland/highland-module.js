// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import highland from 'highland';

export default angular
  .module('highland', [])
  .value('Î»', highland)
  .value('highland', highland).name;
