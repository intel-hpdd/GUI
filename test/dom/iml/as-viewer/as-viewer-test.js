import highland from 'highland';
import * as fp from 'intel-fp';
import broadcaster from '../../../../source/iml/broadcaster.js';
import asViewerDirective from '../../../../source/iml/as-viewer/as-viewer.js';

describe('as viewer', () => {
  let $compile, $scope, el, s, getText, v;

  beforeEach(
    module($compileProvider => {
      $compileProvider.directive('asViewer', asViewerDirective);
    })
  );

  describe('with transform', () => {
    beforeEach(
      inject(($rootScope, _$compile_) => {
        $compile = _$compile_;

        const template = `
        <div as-viewer stream="stream" args="args" transform="add1(stream, args)">
          <span class="num" ng-init="setNum(viewer)">{{ num }}</span>
        </div>
      `;

        $scope = $rootScope.$new();

        s = highland();
        $scope.stream = broadcaster(s);

        $scope.args = [2];

        $scope.add1 = function add1(s, args) {
          return s.map(highland.add.apply(highland, args));
        };

        $scope.setNum = function setNum(s) {
          v = s;
          spyOn(v, 'destroy').and.callThrough();
          v.each(function(x) {
            $scope.num = x;
          });
        };

        el = $compile(template)($scope);
        $scope.$digest();

        const find = el[0].querySelector.bind(el[0]);
        getText = fp.flow(find, fp.view(fp.lensProp('textContent')));
      })
    );

    it('should add 2 to num', () => {
      s.write(1);
      $scope.$digest();

      expect(getText('.num')).toEqual('3');
    });

    it('should destroy the viewer when scope is destroyed', () => {
      $scope.$destroy();

      expect(v.destroy).toHaveBeenCalled();
    });
  });

  describe('without transform', function() {
    beforeEach(
      inject(function($rootScope, _$compile_) {
        $compile = _$compile_;

        const template = `
        <div>
          <div as-viewer stream="stream">
            <span class="a" ng-init="setA(viewer)">{{ a }}</span>
          </div>
          <div as-viewer stream="stream">
            <span class="b" ng-init="setB(viewer)">{{ b }}</span>
          </div>
      </div>
    `;

        $scope = $rootScope.$new();
        s = highland();
        $scope.stream = broadcaster(s);

        $scope.setA = function setA(s) {
          s.each(function(x) {
            $scope.a = x.a;
          });
        };

        $scope.setB = function setB(s) {
          s.each(function(x) {
            $scope.b = x.b;
          });
        };

        el = $compile(template)($scope);
        $scope.$digest();

        const find = el[0].querySelector.bind(el[0]);
        getText = fp.flow(find, fp.view(fp.lensProp('textContent')));
      })
    );

    describe('multiple children', function() {
      beforeEach(function() {
        s.write({
          a: 'eeey',
          b: 'bee'
        });
        $scope.$digest();
      });

      it('should set a', function() {
        expect(getText('.a')).toEqual('eeey');
      });

      it('should set b', function() {
        expect(getText('.b')).toEqual('bee');
      });

      describe('on destroy b', function() {
        beforeEach(function() {
          $scope.$$childTail.$destroy();
          s.write({
            a: 'a',
            b: 'b'
          });
          $scope.$digest();
        });

        it('should update a', function() {
          expect(getText('.a')).toEqual('a');
        });

        it('should not update b', function() {
          expect(getText('.b')).toEqual('bee');
        });
      });
    });

    describe('adding a child', function() {
      beforeEach(function() {
        const template = '<div as-viewer stream="stream">\
        <span class="c" ng-init="setC(viewer)">{{ c }}</span>\
      </div>';

        $scope.setC = function setC(s) {
          s.each(function(x) {
            $scope.c = x.c;
          });
        };

        const child = $compile(template)($scope);
        el[0].appendChild(child[0]);
        s.write({
          a: 'a',
          b: 'b',
          c: 'c'
        });
        $scope.$digest();
      });

      it('should update a', function() {
        expect(getText('.a')).toEqual('a');
      });

      it('should update b', function() {
        expect(getText('.b')).toEqual('b');
      });

      it('should update c', function() {
        expect(getText('.c')).toEqual('c');
      });
    });
  });
});
