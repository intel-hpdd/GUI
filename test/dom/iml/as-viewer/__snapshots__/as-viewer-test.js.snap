// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as viewers with proper args should match the snapshot 1`] = `
<div>
  <div
    id="multiple-viewers-component"
  >
    <h1
      id="name"
    >
      test-component
    </h1>
    <div
      id="key1"
    >
      val1
    </div>
    <div
      id="key2"
    >
      dewey!
    </div>
    <div
      id="key3"
    >
      val3
    </div>
  </div>
</div>
`;
